client/finality-grandpa/src/environment.rs

```diff
@@ -924,22 +929,133 @@ where
 	fn prevote_equivocation(
 		&self,
 		_round: RoundNumber,
-		equivocation: ::finality_grandpa::Equivocation<Self::Id, Prevote<Block>, Self::Signature>
+		equivocation: finality_grandpa::Equivocation<Self::Id, Prevote<Block>, Self::Signature>,
 	) {
 		warn!(target: "afg", "Detected prevote equivocation in the finality worker: {:?}", equivocation);
-		// nothing yet; this could craft misbehavior reports of some kind.
+		match report_equivocation(
```

Comment:
Why not have `report_equivocation` a method within an `Environment` `impl`? Would save us on passing the first 3 arguments.

Reply:
Done.